### PR TITLE
Drop terraform remote, use ansible s3 tasks

### PR DESCRIPTION
**Pull request by spiffxp**
See individual commits for details

### CODE REVIEW

File: ansible/roles/ci-properties/defaults/main.yaml [hunk 1]
```diff
@@ -12,19 +12,20 @@ ci_container_name: jenkins
 ci_container_path: "{{ containers_dir }}/jenkins"
 ci_container_tag: samsung_ag/jenkins-server
 ci_email: "maratoid+samsungjenkins@gmail.com"
-ci_hostname: "{{ aws_prefix }}.kubeme.io"
+ci_hostname: "{{ ci_name }}.kubeme.io"
 ci_jobs_repo: https://github.com/samsung-cnct/kraken-ci-jobs
 ci_jobs_repo_branch: master
+ci_name: "{{ aws_prefix }}"
```
Comment (maratoid) [inline ansible/roles/ci-properties/defaults/main.yaml]: So, why add another thing that is the same as aws_prefix?

Comment (spiffxp) [inline ansible/roles/ci-properties/defaults/main.yaml]: aws_prefix makes sense for aws elements, but we were using it for non-aws stuff as well

I'd prefer to invert, and have ci_name be what we pass in and derive aws_prefix from that, but seemed out of scope

Comment (maratoid) [inline ansible/roles/ci-properties/defaults/main.yaml]: But you also replaced aws_prefix with ci_name for a bunch of aws resources.

aws_secgroup_name, aws_key_name and so on

File: ansible/roles/terraform/tasks/terraform-run.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    -refresh=false
+    -no-color
+    -input=false
+    -state={{ terraform_instance_dir }}/terraform.tfstate
+    -var-file={{ terraform_instance_dir }}/terraform.tfvars
+    chdir={{ terraform_dir }}
+  ignore_errors: yes
+  register: terraform_result_b
+
+- name: Dump terraform stderr on failure
```
Comment (maratoid) [inline ansible/roles/terraform/tasks/terraform-run.yaml]: Can we avoid duplicating this by registering the same result variable for both?

File: ansible/roles/terraform/tasks/terraform-run.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  fail: msg="Terraform {{ terraform_mode }} failed with:\n{{ terraform_result_b.stderr }}"
+  when: terraform_result_b|failed
```
Comment (maratoid) [inline ansible/roles/terraform/tasks/terraform-run.yaml]: It think we should still push state if terraform failed on creation or destruction. If the state changed, it should be pushed to s3.

Comment (maratoid) [inline ansible/roles/terraform/tasks/terraform-run.yaml]: If you end up disagreeing - then there is no need for all the error handling. We can just fail.

File: ansible/roles/ci-properties/defaults/main.yaml (hunk 1 above)
Comment (spiffxp) [inline ansible/roles/ci-properties/defaults/main.yaml]: will alter setup.sh/utils.sh to drop aws_prefix entirely, just pass ci_name instead

File: ansible/roles/terraform/tasks/terraform-run.yaml (hunk 2 above)
Comment (spiffxp) [inline ansible/roles/terraform/tasks/terraform-run.yaml]: unfortunately, skipped tasks still (re)register their variables, overwriting previous values

File: ansible/roles/terraform/tasks/terraform-run.yaml (hunk 3 above)
Comment (spiffxp) [inline ansible/roles/terraform/tasks/terraform-run.yaml]: yeah just going to drop the handling, we'll let ansible fail fast

**Comment by maratoid** [comment]
lgtm, comments pending

File: README.md [hunk 4]
```diff
@@ -2,37 +2,42 @@
...
+    $ . .env-testpipe && ./setup.sh --ci-name testpipe --dump-data yes
 
 ### Try it out
 
 Point your browser to
 
-    https://pipelet.kubeme.io
+    https://testpipe.kubeme.io
 
 After a self-signed cert SSL warning you should see the jenkins dashboard. Now try:
```
Comment (maratoid) [inline README.md]: maybe a note on test certs vs real letsencrypt certs?

Comment (maratoid) [inline README.md]: so, now that we default to non-test certs, perhaps a note on the --test-instance switch?

**Comment by maratoid** [comment]
lgtm, pending the readme note.

**Comment by maratoid** [comment]
mergeit

**Pull request merged**